src/Schema/Events/ImpressionTracking.ts

```diff
@@ -80,3 +80,28 @@ export interface TooltipViewed {
   context_owner_type: PageOwnerType
   type: string
 }
+
+/**
+ * A user sees a an error message
+ *
+ * This schema describes events sent to Segment from [[ErrorMessageViewed]].
+ *
+ *  @example
+ *  ```
+ *  {
+ *    action: "errorMessageViewed",
+ *    context_owner_type: "orders",
+ *    context_owner_id: "order-id",
+ *    label: "This is an error",
+ *    error_code: "charge_authorization_failed"
+ *  }
+ * ```
+ *
+ */
+export interface ErrorMessageViewed {
+  action: ActionType.errorMessageViewed
+  context_owner_id: string
+  context_owner_type: OwnerType
+  label: string
```

Comment:
We have a title and a message field, both of which are interested. Could we have two fields for those, or should we smoosh them together as one label?

Reply:
Instead of using the error message as the label, we decided to describe the user action that led to the error (ex. user submitted payment method) so that we can differentiate the places where an error dialog can be displayed in one route.

Reply:
How do you call this "action" field?

Reply:
We can have a property for each: message / title, I'll add

Reply:
That works too. We thought that a field to describe the action that caused an error would be better than the title/message pair because:

1. The same title/message pair can be shown to users for different actions on the same page, so mapping an event to the action will allow us to pinpoint the action that caused it.
2. TX is planning to revise the copy of existing error dialogs in the checkout flow, so we could lose track of the events that we have collected if we rely on the title/message pair.

I'm not sure what I would call this field. I did a quick search through the existing Cohesion types and didn't find one that matched. Maybe something like `context_user_action`?

Reply:
I can add an `action` property to record also that specific action.